networkx/generators/threshold.py

```diff
@@ -307,7 +307,7 @@ def threshold_graph(creation_sequence, create_using=None):
     while ci:
         (v,node_type)=ci.pop(0)
         if node_type=='d': # dominating type, connect to all existing nodes
-            for u in G.nodes(): 
+            for u in list(G):
```

Comment:
@dschult This will throw a RuntimeError `RuntimeError: dictionary changed size during iteration`. We add a edge to the graph in every iteration so I think `for u in list(G):` should be used instead of `for u in G:`